ci/lava/helpers.py

```diff
@@ -0,0 +1,240 @@
+                "artifactory-proxy.mbed-linux.arm.com into /etc/hosts."
+            )
+            try:
+                f = open("/etc/hosts", "a")
+                f.write(
+                    "192.168.130.43  artifactory-proxy.mbed-linux.arm.com\n"
+                )
+            except Exception as inst:
+                print("\n\nError {}\n\n".format(type(inst)))
+                print("Could not update /etc/hosts. Perhaps run as root?")
+
+            # re-try the urlretrieve.
+            try:
+                urllib.request.urlretrieve(url, filename)
+            except Exception as inst:
+                print("\n\nError {}\n\n".format(type(inst)))
+                print("\n\nAlternative method also failed.\n\n")
+                filename = None
+    return filename
+
+
+def get_dut_address(dut, execute_helper):
+    """Calculate the DUT IP address using mbl-cli."""
+    dut_addr = ""
+
+    exit_code, output, error = execute_helper.execute_command(
+        ["mbl-cli", "list"]
+    )
+
+    # The default value for dut is set to auto. This means use the first
+    # device returned by the mbl-cli. In the test farm set up there is a
+    # one-to-one mapping of device to container, so this is sufficient.
+    # However, allowing a dut to be specified allows manual testing when
+    # there is more than one device detectable on the network.
+    #
+    # The mbl-cli list command returns data in the following format:
+    #     Discovering devices. This will take up to 30 seconds.
+    #     1: mbed-linux-os-2075: fe80::21f:7bff:fe86:a738%enp0s31f6
+    #     2: mbed-linux-os-158: fe80::ecfb:3fff:fece:9376%enp0s20f0u14u1
+    #
+    # The regular expressions below extract the IP address and interface
+    # name as that is used to communicate to the DUT.
+
+    if dut == "auto":
+        # Take the first found device and use that
+        pattern = re.compile(r"\s*1:[^:]+:\s+(?P<address>\S+)")
+        for line in output.splitlines():
+            match = pattern.match(line)
+            if match:
+                dut_addr = match.group("address")
+                break
+    else:
+        # Match a device against the provided dut parameter
+        pattern = re.compile(r"\s*.:[^:]+:\s+(?P<address>\S+)")
+        for line in output.splitlines():
+            if dut in line:
+                match = pattern.match(line)
+                if match:
+                    dut_addr = match.group("address")
+                    break
+    return dut_addr
+
+
+def get_kernel_version(dut_addr, execute_helper):
+    """Get the kernel version running on the DUT using mbl-cli."""
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", 'sh -l -c "uname -a"'], dut_addr
+    )
+    return exit_code, output
+
+
+def strings_grep(dut_addr, execute_helper, file_path, pattern):
+    """Run strings command on a file and grep for a pattern using mbl-cli."""
+    command = "strings {} | grep {}".format(file_path, pattern)
```

Comment:
Should we be careful about making sure these `grep` patterns are properly escaped? I'm not sure.